### PR TITLE
backport calico charts, most important adopt calico-node liveness pro…

### PR DESCRIPTION
**Pull request by majst01**
…be execution, nodeselector now out of beta

**What this PR does / why we need it**:

We found some differences between the charts shipped here and upstream calico. I changed:
- calico-node liveness probe
- calico-node got one more rbac rule
- enlarge calico-node cpu requests

```

### CODE REVIEW

File: charts/internal/calico/templates/kube-controller/deployment-calico-kube-controllers.yaml [hunk 1]
```diff
@@ -32,7 +32,7 @@ spec:
         gardener.cloud/role: system-component
     spec:
       nodeSelector:
-        beta.kubernetes.io/os: linux
+        kubernetes.io/os: linux
```
Comment (rfranzke) [inline charts/internal/calico/templates/kube-controller/deployment-calico-kube-controllers.yaml]: Will this work on 1.10 clusters?

Comment (majst01) [reply charts/internal/calico/templates/kube-controller/deployment-calico-kube-controllers.yaml]: 1.10 ? still exists, unsure, will revert this if 1.10 is required to work as well

Comment (majst01) [reply charts/internal/calico/templates/kube-controller/deployment-calico-kube-controllers.yaml]: reverted

**Comment by zanetworker** [comment]
Hi @majst01, 

I have a few questions: 

- why do we need to support kubeadm use-cases? 
- by upstream you mean the master branch or 3.11.2?

**Comment by majst01** [comment]
No need to suport kubeadm, but i feel it difficult to cherry pick parts of the charts, instead of using complete upstream. 
I compared against 3.13 most recent stable.

**Comment by zanetworker** [comment]
@majst01 currently we support 3.11.2, thats why the config is different.

Does it make sense to file a different PR that upgrades calico to 3.13, which also includes updating the images to comply with the YAMLs?

**Comment by majst01** [comment]
Yes sure, i was a bit afraid of doing so in one go, but i can do this.

**Comment by majst01** [comment]
Hi @zanetworker new PR is https://github.com/gardener/gardener-extension-networking-calico/pull/20

**Pull request closed without merging**